src/amoc_distribution/amoc_cluster.erl

```diff
@@ -282,3 +280,8 @@ maybe_set_master(Node, #state{new_connection_action = Action}) ->
     %% to avoid a possibility of the amoc_cluster deadlock while
     %% running the Action call set_master_node/2 asynchronously
     spawn(fun() -> set_master_node(Node, Action) end).
+
+-spec execute_nodes(atom(), [node()], #{any() => any()}) -> ok.
+execute_nodes(Name, Nodes, State) ->
```

Comment:
poor nodes, what have they done so that now they have to be executed?
please rename it into execute_nodes_event or something like that.